openqaoa/workflows/optimizer.py

```diff
@@ -713,7 +713,11 @@ def optimize(self, verbose=False):
             spin_map = rqaoa.spin_mapping(problem, max_terms_and_stats)
             # Eliminate spins and redefine problem
             new_problem, spin_map = rqaoa.redefine_problem(problem, spin_map)
-
+            # In case eliminations cancel out the whole graph, break the loop (solve classically) before reaching the predefined cutoff.
+            if new_problem == problem:
```

Comment:
Is this always going to be correct? I guess the constant term may change, or what happens if symmetric relabeling happens?

Reply:
It comes from the rqaoa.py file: I'm passing the old problem only if the new problem dict is empty.What do you mean by symmetric relabeling?

https://github.com/entropicalabs/openqaoa/blob/851352621182aa73ef998f722f5da9fa7551b486/openqaoa/rqaoa/rqaoa.py#L502

Reply:
Assume the old problem has [0,1] and corresponding weight [1], and the new problem has [1,0] and [1]. Wouldn't this be the same graph? But the equality would fail.

Don't focus on my example: I was just wondering if the code would break in cases where you have, for example equivalent graphs but different objects

Reply:
I'd say no because the way the code is written the problem is always defined in increasing order of the nodes. But will discuss with Eze and Vishal after the holidays, very likely there are edge cases I haven't considered.

Reply:
@Q-lds Yes, @kidiki, you are right. This was an issue Eze and I faced before and decided to represent the nodes in increasing order.